star-wars/src/LinkButton.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { withRouter } from 'react-router'
+
+//LinkButton class copied from https://stackoverflow.com/a/49439893/284830
+const LinkButton = (props) => {
+  const {
+    history,
+    location,
+    match,
+    staticContext,
+    to,
+    onClick,
+    // ⬆ filtering out props that `button` doesn’t know what to do with.
```

Comment:
This is a common practice, and it's usually why developers tend to be explicit with the props that are being spread into a component. But when props have to be spread, just destructure what you don't want 😄